third-party/re2/Makefile

```diff
@@ -21,6 +21,27 @@ CHPL_RE2_CXXFLAGS=-mmic
 endif
 endif
 
+
+# re2's makefile throws gcc specific flags that are not supported by the cray
+# compiler. This is intended to effectively replace the re2 makefile lines:
+#
+#     CXXFLAGS?=-Wall -O3 -g -pthread # can override
+#     RE2_CXXFLAGS?=-Wsign-compare -c -I. $(CCPCRE)  # required
+#     LDFLAGS?=-pthread
+#
+# with:
+#
+#     CXXFLAGS?= -O3
+#     RE2_CXXFLAGS?= -c -I. $(CCPCRE)
+#     LDFLAGS?=
+ifeq ($(CHPL_MAKE_TARGET_COMPILER),cray-prgenv-cray)
+CHPL_RE2_CXXFLAGS=-O3
+CHPL_RE2_CFG_OPTIONS += LDFLAGS=
+CHPL_RE2_CFG_OPTIONS += 'RE2_CXXFLAGS= -c -I. $$(CCPCRE)'
+endif
+
+CHPL_RE2_CFG_OPTIONS += $(CHPL_RE2_MORE_CFG_OPTIONS)
```

Comment:
It's not defined anywhere. We have the *MORE_CFG_OPTIONS for most of our thirdparty makefiles so that we can set additional config options without changing the makefiles. i.e. 
`export CHPL_HWLOC_MORE_CFG_OPTIONS=--disable-libnuma`